src/charm.py

```diff
@@ -24,37 +27,37 @@ def __init__(self, *args: typing.Any):
             args: Arguments passed to the CharmBase parent constructor.
         """
         super().__init__(*args)
-        self.irc = IRCService()
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
-        self.framework.observe(self.on.install, self._on_install)
-        self.framework.observe(self.on.start, self._on_start)
+        self._irc = IRCService()
+        self._database = DatabaseService(self, DATABASE_RELATION_NAME)
+        self.framework.observe(
+            self._database.database.on.database_created, self.reconcile
+        )
+        self.framework.observe(
+            self.on[DATABASE_RELATION_NAME].relation_broken,
+            self.reconcile
+        )
+        self.framework.observe(self.on.config_changed, self.reconcile)
+        self.framework.observe(self.on.install, self.reconcile)
+        self.framework.observe(self.on.start, self.reconcile)
```

Comment:
Even if we endup calling a reconcile method for every events, I'm in favor of keeping the hook methods following our conventions